AsyncDisplayKit/ASVideoNode.h

```diff
@@ -0,0 +1,34 @@
+
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+
+typedef NS_ENUM(NSUInteger, ASVideoGravity) {
+  ASVideoGravityResizeAspect,
+  ASVideoGravityResizeAspectFill,
+  ASVideoGravityResize
+};
+
+@protocol ASVideoNodeDelegate;
+
+@interface ASVideoNode : ASControlNode<_ASDisplayLayerDelegate>
+@property (atomic, strong, readwrite) AVAsset *asset;
+@property (atomic, strong, readonly) AVPlayer *player;
+@property (atomic, strong, readonly) AVPlayerItem *currentItem;
```

Comment:
I'm not that familiar with the different ways AVPlayerItems are used in practice, for looping videos etc, but I'm wondering if it is the right thing to have this called currentItem.  If it changes, maybe there should be a delegate method to signal that, and/or this should be an array of items?  It's important that we don't have to change the API surface later, only add to it.

Reading the documentation on AVPlayerItem yields much interesting information.  We should make very sure that the API we expose allows for asynchronously loading the key information.  For example if someone creates a node with an asset and then immediately calls for the currentItem, do we have a way to make sure we don't block waiting for that information to be parsed out of the video hosted on a remote server?  Even once the video is local, it can take surprisingly long (in terms of frame dropping times, on older devices) to parse information such as the duration from videos, so we should try to insulate users from that "forcibly" via the API design.  This may mean no property for the item at all, and only a method like currentItemWithCompletionBlock: or something (not a great name - think about it once you've learned more about the async fetching options from the docs).  Example snippet:

"Discussion
The value of each key in automaticallyLoadedAssetKeys will be automatically be loaded by the underlying AVAsset before the receiver achieves the status AVPlayerItemStatusReadyToPlay; i.e. when the item is ready to play, the value returned by invoking the asset property’s statusOfValueForKey:error: method will be one of the terminal status values, either AVKeyValueStatusLoaded, AVKeyValueStatusFailed, or AVKeyValueStatusCancelled.

IMPORTANT
The asset property keys @"playable" and @"compatibleWithSavedPhotosAlbum" are not eligible for automatic loading by AVPlayerItem. You must use the AVAsset method loadValuesAsynchronouslyForKeys:completionHandler: to load the values of those properties asynchronously."